cmake/Dependencies.cmake

```diff
@@ -1260,6 +1260,7 @@ if(USE_ROCM)
     list(APPEND HIP_CXX_FLAGS -fPIC)
     list(APPEND HIP_CXX_FLAGS -D__HIP_PLATFORM_HCC__=1)
     list(APPEND HIP_CXX_FLAGS -DCUDA_HAS_FP16=1)
+    list(APPEND HIP_CXX_FLAGS -DUSE_ROCM)
```

Comment:
I think this change is not required.

Reply:
I would normally agree with you, but I went through this with Jeff and Jithun.  Without this, hipCachingAllocator.cpp won't receive the USE_ROCM flag, and in turn won't enable the graph functionality in the caching allocator and we will put graphs into the general allocation pool, leading to wonky results.

I'll chat with you in more detail if you need!  This one I wasn't thrilled with, but nothing more elegant introduced itself.  I think it's related to us never needing USE_ROCM in c10 code.